chart_review/commands/mcnemar.py

```diff
@@ -0,0 +1,109 @@
+"""Methods for McNemar calculations."""
+
+import argparse
+from typing import Optional
+
+import rich
+import rich.box
+import rich.table
+import rich.text
+from scipy.stats import binom, chi2
+
+from chart_review import cli_utils, console_utils
+
+
+def make_subparser(parser: argparse.ArgumentParser) -> None:
+    cli_utils.add_project_args(parser)
+    cli_utils.add_output_args(parser)
+    parser.add_argument("truth_annotator")
+    parser.add_argument("annotator1")
+    parser.add_argument("annotator2")
+    parser.add_argument("--continuity-correction", action="store_true")
+    parser.set_defaults(func=print_mcnemar)
+
+
+def print_mcnemar(args: argparse.Namespace) -> None:
+    """
+    McNemar calculation between three annotators (one truth, two normal).
+
+    More information:
+     https://en.wikipedia.org/wiki/McNemar's_test
+     https://pmc.ncbi.nlm.nih.gov/articles/PMC3716987/
+    """
+    reader = cli_utils.get_cohort_reader(args)
+    truth = args.truth_annotator
+    annotator1 = args.annotator1
+    annotator2 = args.annotator2
+
+    all_people = {truth, annotator1, annotator2}
+    annotators = [annotator1, annotator2]
+
+    for annotator in all_people:
+        if annotator not in reader.note_range:
+            raise ValueError(f"Unrecognized annotator '{annotator}'")
+
+    if len(all_people) != 3:
+        raise ValueError("Can’t compare the same annotator with themselves.")
+
+    # Grab the intersection of ranges
+    note_range = reader.note_range[truth]
+    for annotator in annotators:
+        note_range &= reader.note_range[annotator]
+
+    labels = [None, *sorted(reader.class_labels, key=str.casefold)]
+
+    # Calculate contingency tables
+    matrices = {
+        label: reader.contingency_table(truth, annotator1, annotator2, note_range, label)
+        for label in labels
+    }
+
+    console = rich.get_console()
+
+    table = cli_utils.create_table(
+        "McNemar", "P-value", "BC", "OL", "OR", "BW", "Label", dense=True
+    )
+    empty_val = "" if args.csv else "N/A"
+    for label in labels:
+        m = matrices[label]
+        mcn, pval = _mcnemar(len(m["OL"]), len(m["OR"]), continuity_correction=True)
+        table.add_row(
+            empty_val if mcn is None else str(round(mcn, 3)),
+            str(round(pval, 3)),
```

Comment:
My only real feedback is here: In cases where pvalues are very small (and luckily in our case, they are!) they are presented as 0. One fix is to use scientific notation, but we don't want to do that for all numbers - that makes our tables look really wonky.

My proposal would be a condition scientific formatting method that returns str(round(mcn, 3)) if that value isn't equal to 0; if rounding gives us a number equal to 0, we uses string formatting: `f'{num:.2e}'`

Let me know what you think!

Reply:
Done.